openpype/hosts/max/plugins/publish/collect_tycache_attributes.py

```diff
@@ -1,76 +1,66 @@
-                         OpenPypePyblishPluginMixin):
+class CollectTyFlowData(pyblish.api.InstancePlugin,
+                        AYONPyblishPluginMixin):
     """Collect Channel Attributes for TyCache Export"""
 
-    order = pyblish.api.CollectorOrder + 0.02
+    order = pyblish.api.CollectorOrder + 0.005
     label = "Collect tyCache attribute Data"
     hosts = ['max']
-    families = ["tycache"]
+    families = ["tyflow"]
+    validate_tycache_frame_range = True
 
-    def process(self, instance):
-        attr_values = self.get_attr_values_from_data(instance.data)
-        attributes = {}
-        for attr_key in attr_values.get("tycacheAttributes", []):
-            attributes[attr_key] = True
-
-        for key in ["tycacheLayer", "tycacheObjectName"]:
-            attributes[key] = attr_values.get(key, "")
+    @classmethod
+    def apply_settings(cls, project_settings):
 
-        # Collect the selected channel data before exporting
-        instance.data["tyc_attrs"] = attributes
-        self.log.debug(
-            f"Found tycache attributes: {attributes}"
+        settings = (
+            project_settings["max"]["publish"]["ValidateTyCacheFrameRange"]
         )
+        cls.validate_tycache_frame_range = settings["active"]
+
+    def process(self, instance):
+        context = instance.context
+        container = rt.GetNodeByName(instance.data["instance_node"])
+        tyc_product_names = [
+            name for name
+            in container.modifiers[0].AYONTyCacheData.tyc_exports
+        ]
+        attr_values = self.get_attr_values_from_data(instance.data)
+        for tyc_product_name in tyc_product_names:
+            self.log.debug(
+                f"Creating instance for operator:{tyc_product_name}")
+            tyc_instance = context.create_instance(tyc_product_name)
+            tyc_instance[:] = instance[:]
+            tyc_instance.data.update(copy.deepcopy(dict(instance.data)))
+            tyc_instance.data["name"] = "{}_{}".format(
+                instance.data["tyc_exportMode"], tyc_product_name)
+            tyc_instance.data["label"] = "{}_{}".format(
+                instance.data["tyc_exportMode"], tyc_product_name)
+            tyc_instance.data["family"] = instance.data["tyc_exportMode"]
+            tyc_instance.data["productName"] = tyc_product_name
+            tyc_instance.data["productType"] = instance.data["tyc_exportMode"]
+            tyc_instance.data["exportMode"] = (
+                2 if instance.data["tyc_exportMode"] == "tycache" else 6
+            )
+            tyc_instance.data["families"] = [instance.data["tyc_exportMode"]]
+            tyc_instance.data["creator_identifier"] = (
+                "io.openpype.creators.max.{}".format(
+                    instance.data["tyc_exportMode"])
+            )
+            tyc_instance.data["publish_attributes"] = {
+                "ValidateTyCacheFrameRange": {"active": attr_values.get(
+                    "has_frame_range_validator")}}
```

Comment:
It may be just me - but I prefer this for readability:

```suggestion
        export_mode = instance.data["tyc_exportMode"]
        tyc_instance.update({
            "name": f"{export_mode}_{tyc_product_name}",
            "label": f"{export_mode}_{tyc_product_name}",
            "family": export_mode,
            "families": [export_mode],
            "productName": tyc_product_name
            "productType": export_mode,
            "exportMode": (
                2 if instance.data["tyc_exportMode":= "tycache" else 6
            ),
            "creator_identifier": f"io.openpype.creators.max.{export_mode}",
            "publish_attributes": {
                "ValidateTyCacheFrameRange": {
                    "active": attr_values.get("has_frame_range_validator")}
            }
        })
```